ethpm_types/manifest.py

```diff
@@ -165,11 +165,12 @@ def check_both_version_and_name(cls, values):
 
     @root_validator
     def check_contract_source_ids(cls, values):
-        if values.get("contract_types") is not None and values.get("sources") is not None:
-            for alias in values["contract_types"]:
-                source_id = values["contract_types"][alias].source_id
-                if source_id and (source_id not in values["sources"]):
-                    raise ValueError(f"'{source_id}' missing from `sources`.")
+        contract_types = values.get("contract_types", {}) or {}
+        for alias in contract_types:
+            source_id = values["contract_types"][alias].source_id
+            sources = values.get("sources", {}) or {}
```

Comment:
```suggestion
            sources = values.get("sources", {})
```

Reply:
`sources` is also `None`-able; this would mean this variable `sources` can potentially be a dict or `None`.
It easiest to just handle that all in one line like this. `mypy` catches it too